src/librbd/AioObjectRequest.cc

```diff
@@ -84,6 +85,9 @@ AioObjectRequest<I>::AioObjectRequest(ImageCtx *ictx, const std::string &oid,
   RWLock::RLocker snap_locker(m_ictx->snap_lock);
   RWLock::RLocker parent_locker(m_ictx->parent_lock);
   compute_parent_extents();
+
+  if (trace && trace->valid())
+    m_trace = ZTracer::Trace(*trace);
```

Comment:
Minor: can't the assignment operate be used directly? or is the copy-constructor doing something else?